apis/kueue/v1beta1/workload_types.go

```diff
@@ -367,6 +367,38 @@ type WorkloadStatus struct {
 	//
 	// +optional
 	AccumulatedPastExexcutionTimeSeconds *int32 `json:"accumulatedPastExexcutionTimeSeconds,omitempty"`
+
+	// waitForPodsReadyRequeuesByReason holds the re-queue state for WaitForPodsReady.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=reason
+	// +patchStrategy=merge
+	// +patchMergeKey=reason
+	WaitForPodsReadyRequeuesByReason []WaitForPodsReadyRequeue `json:"waitForPodsReadyRequeuesByReason,omitempty"`
```

Comment:
Why do we need to preserve this information in Workload? Why do we not just increment the Prometheus metrics count?

Reply:
According to the description ("The number of workloads evicted due to hitting one of the WaitForPodsReady timeouts at least once"), we need to track whether this is the first time the timeout has occurred for a given workload. If it’s the first occurrence, we can safely increment the metric.

Reply:
Yes, the idea is to count workloads with the timeout, not necessarily how many times a timeout was hit.

Reply:
It makes sense for a Workload to be counted only once, since if it failed once it's likely to fail again, and then it adds noise to the metric. I believe the main goal of this metric should be to track how many workloads have failed, but we shouldn't double/triple count it